### PR TITLE
Structured Text language support

### PR DESCRIPTION
**Pull request by Serhioromano**
New PR with clean Prism fork

### CODE REVIEW

File: components.json [hunk 1]
```diff
@@ -458,6 +458,10 @@
 			"title": "Icon",
 			"owner": "Golmote"
 		},
+		"iecst": {
+			"title": "Language Title",
+			"owner": "Your GitHub name"
```
Comment (RunDevelopment) [inline components.json:463]: Is that really your name?

File: components/prism-iecst.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+			lookbehind: true,
+			greedy: true,
+		},
+	],
+	string: {
+		pattern: /(["'])(?:\\(?:\r\n|[\s\S])|(?!\1)[^\\\r\n])*\1/,
+		greedy: true,
+	},
+	"class-name": /\b(?:END_)?(?:PROGRAM|CONFIGURATION|INTERFACE|FUNCTION_BLOCK|FUNCTION|ACTION|TRANSITION|TYPE|STRUCT|(?:INITIAL_)?STEP|NAMESPACE|LIBRARY|CHANNEL|FOLDER|RESOURCE|VAR_(?:GLOBAL|INPUT|PUTPUT|IN_OUT|ACCESS|TEMP|EXTERNAL|CONFIG)|VAR|METHOD|PROPERTY)\b/i,
+	keyword: /\b(?:(?:END_)?(?:IF|WHILE|REPEAT|CASE|FOR)|ELSE|FROM|THEN|ELSIF|DO|TO|BY|PRIVATE|PUBLIC|PROTECTED|CONSTANT|RETURN|EXIT|CONTINUE|GOTO|JMP|AT|RETAIN|NON_RETAIN|TASK|WITH|UNTIL|USING|EXTENDS|IMPLEMENTS|GET|SET|__TRY|__CATCH|__FINALLY|__ENDTRY)\b/,
```
Comment (RunDevelopment) [inline components/prism-iecst.js:18]: We use single quotes for all token names. This makes them visually distinct from other JavaScript properties (e.g. `pattern`).

File: components/prism-iecst.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	variable: /\b(?:AT|BOOL|BYTE|(?:D|L)?WORD|U?(?:S|D|L)?INT|L?REAL|TIME(?:_OF_DAY)?|TOD|DT|DATE(?:_AND_TIME)?|STRING|ARRAY|ANY|POINTER)\b/,
+	symbol: /%[IQM][XBWDL][\d.]*|%[IQ][\d.]*/,
+	number: /\b(?:16#[\da-f]+|2#[01_]+|0x[\da-f]+)\b|\b(?:T|D|DT|TOD)#[\d_shmd:]*|\b[A-Z]*\#[\d.,_]*|(?:\b\d+\.?\d*|\B\.\d+)(?:e[+-]?\d+)?/i,
+	boolean: /\b(?:TRUE|FALSE|NULL)\b/,
+	function: /\w+(?=\()/,
+	operator: /(?:\+|S?R?:?\=>?|:|\^|\-|&&?|\*\*?|\/|<=?|>=?|OR|AND|MOD|NOT|XOR|LE|GE|EQ|NE|GE|LT)/,
```
Comment (RunDevelopment) [inline components/prism-iecst.js:24]:
```suggestion
	operator: /(?:S?R?:?=>?|&&?|\*\*?|<=?|>=?|[-:^/+])|\b(?:OR|AND|MOD|NOT|XOR|LE|GE|EQ|NE|GE|LT)\b/,
```

File: components.json (hunk 1 above)
Comment (Serhioromano) [reply components.json]: Oopc!

File: components/prism-iecst.js (hunk 2 above)
Comment (Serhioromano) [reply components/prism-iecst.js:18]: done

**Comment by Serhioromano** [comment]
You are obviously good at Regex. Although mine regex works, I see how your suggestions are better. I've learned some new things today.еhank you for taking the time and explaining your changes.

**Comment by Serhioromano** [comment]
Again! What now? I cannot find errors. It was all green. After you request changes, I resolve them all but it always continues to show that I have one change request.

What do I do not? I did not do anything, just changed the language name and accepted your proposals.

**Comment by Serhioromano** [comment]
🥺 Accept it already!!

**Comment by maddie** [comment]
Hmm, some of the comments seem to be missed by the regex:

image

Code: 

```iecst
	iSize: INT;(*发送数据的组数*)
	sTritonDataSend: TritonDataSend;		(*发送的数据的结构体，结构体中发送数据的结构体中的szKey和szValue长度均为64个字节（由于内存长度限制）*)
	sTritonDataRecv: TritonDataRecv;(*从Neptune获取的数据*)
	itriton_status: INT;			（*获取和Neptune连接状态*）
	aTempKey: ARRAY [0..99] OF string;
	aTempNote: ARRAY [0..99] OF string;
	strJsonData: ARRAY [0..9] OF JsonDataString;
	struuid: STRING;				（*获取Triton配置的设备序列号*）
	intuuid: ARRAY [0..99] OF BYTE;	（*获取设备*）
	(*
	  需要在<数据类型>中定义以下结构体：
	  TYPE JsonDataString :
        STRUCT
        szKey:STRING;
        szValue:STRING;
        END_STRUCT
      END_TYPE
	*)
```

File: components/prism-iecst.js (hunk 2 above)
Comment (RunDevelopment) [reply components/prism-iecst.js:18]: Sorry, I meant __all__ token names. Not just ones where you have to use quotes anyway. E.g. `'keyword'` also get's quoted.

**Comment by RunDevelopment** [comment]
> 🥺 Accept it already!!

We are basically there already. Just some formatting and I'll merge this.

**Comment by Serhioromano** [comment]
> iSize: INT;(*发送数据的组数*)
> 	sTritonDataSend: TritonDataSend;		(*发送的数据的结构体，结构体中发送数据的结构体中的szKey和szValue长度均为64个字节（由于内存长度限制）*)
> 	sTritonDataRecv: TritonDataRecv;(*从Neptune获取的数据*)
> 	itriton_status: INT;			（*获取和Neptune连接状态*）
> 	aTempKey: ARRAY [0..99] OF string;
> 	aTempNote: ARRAY [0..99] OF string;
> 	strJsonData: ARRAY [0..9] OF JsonDataString;
> 	struuid: STRING;				（*获取Triton配置的设备序列号*）
> 	intuuid: ARRAY [0..99] OF BYTE;	（*获取设备*）
> 	(*
> 	  需要在<数据类型>中定义以下结构体：
> 	  TYPE JsonDataString :
>         STRUCT
>         szKey:STRING;
>         szValue:STRING;
>         END_STRUCT
>       END_TYPE
> 	*)

Somehow the `(` is not a bracket. You can even see it in the code it is different. I change it to bracket and it works.

**Comment by Serhioromano** [comment]
> > 🥺 Accept it already!!
> 
> We are basically there already. Just some formatting and I'll merge this.

So what else should I do?

**Comment by RunDevelopment** [comment]
[This](https://github.com/PrismJS/prism/pull/2311#discussion_r407222473).

File: components/prism-iecst.js (hunk 2 above)
Comment (Serhioromano) [reply components/prism-iecst.js]: done

**Comment by RunDevelopment** [comment]
Btw. the `（` is a ["FULLWIDTH LEFT PARENTHESIS"](https://www.fileformat.info/info/unicode/char/ff08/index.htm). I don't think ST supports this, does it?

**Comment by RunDevelopment** [comment]
Thank you for contributing @Serhioromano!

**Comment by Serhioromano** [comment]
> Btw. the `（` is a ["FULLWIDTH LEFT PARENTHESIS"](https://www.fileformat.info/info/unicode/char/ff08/index.htm). I don't think ST supports this, does it?

No it doesn't

**Pull request merged**